WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift

```diff
@@ -323,7 +323,8 @@ final class BlazeCampaignCreationFormViewModel: ObservableObject {
 
     func onLoad() async {
         await withTaskGroup(of: Void.self) { group in
-            if suggestions.isEmpty {
+            // Make sure AI Suggestions are called just if suggestions are empty, and we didn't already populated tagline, description and ctaText.
+            if suggestions.isEmpty && tagline.isEmpty && description.isEmpty && ctaText.isEmpty {
```

Comment:
I don't quite understand this part, could you explain why we need the change? From what I understand, `onLoad` is triggered with `task`, i.e. when the creation form appears. There's no other trigger to be prevented.

Reply:
Now, those fields are populated even if the API call fails. Previously, if the API call failed and you navigated to another screen, like moving forward in the navigation, the `loadAISuggestions` would be called again because `suggestions` was empty. However, we now want to avoid calling AI suggestions again if the ad was pre-populated with product information. This prevents unnecessary UI updates, which can be confusing for the user. If the network request initially fails and we populate it with product info, the user might not notice that the ad information was updated with AI suggestions the second time.

Reply:
Gotcha, so this can be triggered again if tapping Confirm Details and then navigating back 👍